internal/models/release.go

```diff
@@ -5,37 +5,53 @@ import (
 )
 
 type Release struct {
-	Name         string    `json:"name"`
-	Created      string    `json:"created"`
-	LastDeployed time.Time `json:"lastDeployed"`
-	Owner        struct {
-		Squad string `json:"squad"`
-		Slack string `json:"slack"`
-	} `json:"owner"`
-	AutoDeploy bool `json:"autoDeploy"`
-	Code       struct {
-		Github string `json:"github"`
-		Ref    string `json:"ref"`
-	} `json:"code"`
-	Build struct {
-		Travis string `json:"travis"`
-	} `json:"build"`
-	Monitoring struct {
-		Datadog struct {
-			Dashboard string `json:"dashboard"`
-			Monitors  string `json:"monitors"`
-		} `json:"datadog"`
-		Sumologic string `json:"sumologic"`
-	} `json:"monitoring"`
-	Artifacts struct {
-		Docker struct {
-			Image string `json:"image"`
-			Tag   string `json:"tag"`
-		} `json:"docker"`
-		Chart struct {
-			Path    string `json:"path"`
-			Version string `json:"version"`
-		} `json:"chart"`
-	} `json:"artifacts"`
-	Status string `json:"status"`
+	Name         string     `json:"name" jsonschema:"description=The name of the release"`
+	Created      time.Time  `json:"created" jsonschema:"description=The time when the release was created"`
+	LastDeployed time.Time  `json:"lastDeployed" jsonschema:"description=The time when the release was last deployed"`
+	Owner        Owner      `json:"owner" jsonschema:"description=Ownership and contact information"`
+	AutoDeploy   bool       `json:"autoDeploy" jsonschema:"description=The state of the release's auto-deployment option"`
+	Code         SourceCode `json:"code" jsonschema:"description=The repository and branch ref of the release's source code"`
+	Build        Build      `json:"build" jsonschema:"description=The CI build page of current release,required=true"`
+	Monitoring   Monitoring `json:"monitoring" jsonschema:"description=The monitoring resources for the release"`
+	Artifacts    Artifacts  `json:"artifacts" jsonschema:"description=The build artifacts of the release"`
+	Status       string     `json:"status" jsonschema:"description=The status of the release,example=deployed,example=failed,example=pending_rollback,example=pending_install,example=pending_upgrade"`
+}
+
+type Owner struct {
```

Comment:
These extra structs add a lot of bloat to the `models` pkg unfortunately. They either have to be public or we need to write a bunch of constructors/setters to allow other packages to construct `Release` values without having access to the internal types.

Reply:
I guess it would be really awkward to construct a `Release` with the anonymous structs as well, we just hadn't gotten that far yet.

Since the `Release` depends on some coming from k8s, github, etc. It might be easier to populate it incrementally instead of all at once (although it's cleaner to do it all at once, just more awkward), so we might want to have a fluent API that allows it to be constructed incrementally, like

```
var rls Release
rls.WithName("name").WithCreated(time.Now()).WithStatus("deployed") // etc
```

Reply:
Thinking about it some more, we can just have public fields with private types, and set fields directly without any fluent/builder API.